flow/scripts/global_route.tcl

```diff
@@ -16,9 +16,36 @@ if {[info exist env(FASTROUTE_TCL)]} {
   }
 }
 
-global_route -guide_file $env(RESULTS_DIR)/route.guide \
-               -congestion_report_file $env(REPORTS_DIR)/congestion.rpt \
-               {*}[expr {[info exists ::env(GLOBAL_ROUTE_ARGS)] ? $::env(GLOBAL_ROUTE_ARGS) : {-congestion_iterations 20 -verbose}}]
+# The default behavior if the user didn't specify GLOBAL_ROUTE_ARGS is to
+# first make an attempt with a few iterations, if this is a hard global
+# route, start from scratch, try harder and logging. The goal here is to make
+# ORFS handle the simple cases without fuzz and to make the harder
+# cases more easily debuggable without having to go spelunking in Tcl.
+#
+# If GLOBAL_ROUTE_ARGS is specified, then we do only what the
+# GLOBAL_ROUTE_ARGS specifies.
+set common_args [list -guide_file $env(RESULTS_DIR)/route.guide \
+                      -congestion_report_file $env(REPORTS_DIR)/congestion.rpt]
+if {[info exists ::env(GLOBAL_ROUTE_ARGS)]} {
+  global_route {*}$common_args {*}$::env(GLOBAL_ROUTE_ARGS)
+} else {
+  for {set i 0} {$i < 2} {incr i} {
+    set congestion_args [list -verbose]
+    if {$i == 1} {
+      global_route {*}$common_args {*}$congestion_args -congestion_iterations 20 -congestion_report_iter_step 1
+      break
+    } else {
+      set result [catch {
+        global_route {*}$common_args {*}$congestion_args -congestion_iterations 5
```

Comment:
If this is the policy we want then I would just implement it in a single run (eg -congestion_report_iter_start 5)

This will only make slow runs slower.

@gadfort @quantamhd thoughts?

Reply:
I would hate to restart the global router from scratch, it seems like adding a flag would be cleaner and better supported across flows.

Reply:
fixed, same as detailed routing, report every 5 iterations, no restarting. Unless GLOBAL_ROUTE_ARGS is specified, in which case do exactly what GLOBAL_ROUTE_ARGS says.

Reply:
@gadfort better?